Add dashboard site documentation

zkamvar opened the pull request:
This adds documentation for the dashboard site: what it does, its architecture, how it's built, and how it's tested. 


Preview: https://hubdocs--326.org.readthedocs.build/en/326/developer/dashboard-site.html

annakrystalli commented on docs/source/developer/dashboard-site.md:
```diff
@@ -0,0 +1,213 @@
+
+(dashboard-site-source)=
+## How the website is built (a tale of two sources)
+
```
I think it would be good to add a short section here called "Configuring the site" that lists what the user is able to configure. 

Then it will also be good to state below that the steps described below are effectively the auto-generated part of the site that takes the user configurations and builds the site from them.

The purpose of the suggestion is to make clear, what the workflow expects as input and is edited by users and what files are then used to auto-generate the site via the builder (which is perfectly described below!)

I know it's alluded to in the table below but I think a short separate section would help conceptually tie the information together better.

zkamvar replied:
Good point. I've added it in commit 41cf7ed9fd4e7d7eb873df70bf11aa8c77bdcedf

annakrystalli commented on docs/source/developer/dashboard-site.md:
```diff
@@ -0,0 +1,213 @@
+   should not need to handle any templating that cannot be easily automated.
+4. We should not have to maintain a server to deploy these sites.
+
+## References
```
I think this section would be better at the end of the topic

zkamvar replied:
done!

annakrystalli commented on docs/source/developer/dashboard-site.md:
```diff
@@ -0,0 +1,213 @@
+
+The images is built using GitHub actions and [deployed to GitHub's container registry](https://github.com/hubverse-org/hub-dash-site-builder/pkgs/container/hub-dash-site-builder/406574245?tag=latest).
+
+We initially cribbed the build workflow from [GitHub's Publishing Docker images
```
What does cribbed mean?

zkamvar replied:
```suggestion
We initially created the build workflow by referencing [GitHub's Publishing Docker images
```

annakrystalli commented on docs/source/developer/dashboard-site.md:
```diff
@@ -0,0 +1,213 @@
+This is similar to the build image except that instead of building the image,
+we are loading it from an artifact and pushing it to the registry.
+
+The final step of this job is to generate an artifact attestation, which is a
```
what does attestation mean? Is it a technical term?

zkamvar replied:
This is a technical term and I think it kinda belongs here since this is developer documentation (not intended for general audiences).

zkamvar replied:
```suggestion
The final step of this job is to generate [an artifact attestation](https://docs.github.com/en/actions/security-for-github-actions/using-artifact-attestations/using-artifact-attestations-to-establish-provenance-for-builds), which is a
```

annakrystalli reviewed (dismissed):
Thank you @zkamvar ! This is great and provides excellent detail! I made some overall minor comments that would have helped myself understand what's going on a little easier.

lshandross commented on docs/source/developer/dashboard-site.md:
```diff
@@ -0,0 +1,262 @@
+1. Combine the contents of `pages/` and `/static/` into a temporary directory, `/tmp/`.
+2. Use `yq` to merge `site-config.yml` into `/tmp/_quarto.yml`.
+3. Update the JavaScript files to point to the correct resources or discard unused files.
+4. Run `quarto render /tmp/` and copy the output to the output directory (`out/`).
```
Do we still have an `out/` directory? I thought it was replaced by `data/`

zkamvar replied:
There is no official directory name here. By default if you do not specify a directory name, it points to `pages/_site`. The reason why that's the default is because that was the behavior before version 1.0.0. The reason that was the behavior before version 1.0.0 is because I had initially written the script to copy the contents of `/site` into `pages` and running quarto on that folder without first creating a temporary folder. By default, quarto dumps the site into a folder called `_site/` at the root of the quarto project, so that ended up being `pages/_site`

lshandross replied:
Ok, can this information be incorporated somehow? I still find the original text a little confusing

zkamvar replied:
I've rearranged the section a bit and added a command line call to the container that hopefully helps.

lshandross replied:
Yes, it looks good now, thank you!

lshandross commented on docs/source/developer/dashboard-site.md:
```diff
@@ -0,0 +1,262 @@
+  yq -->|yq -i '...' _quarto.yml | _quarto.yml
+  _quarto.yml --> quarto
+  contents --> quarto
+  quarto -->|quarto render /tmp/| _site/ --> out/
```
Change `out/` to `data/` here if applicable (i.e., I'm correct in the above comment)

lshandross commented on docs/source/developer/dashboard-site.md:
```diff
@@ -0,0 +1,262 @@
+  trigger.
+
+There is a bit of a dance that's required for this job to run, which is
+described in GitHub's Publishing Docker images guide (see above for link).
```
Can we just link this page again here?

zkamvar replied:
good point! I think I was originally trying to [avoid using duplicate links to the same page](https://www.nngroup.com/articles/duplicate-links/), but you are correct that in this context it makes sense, especially given this guideline from cal tech's [digital accessibility guide on links](https://dap.berkeley.edu/learn/concepts/links)

> If you link to the same thing twice on a page, use the same link text in both places.

```suggestion
described in [GitHub's Publishing Docker images
guide](https://docs.github.com/en/actions/use-cases-and-examples/publishing-packages/publishing-docker-images#publishing-images-to-github-packages).
```

lshandross commented on docs/source/developer/dashboard-site.md:
```diff
@@ -0,0 +1,262 @@
+(dashboard-site-source)=
+## How the website is built (a tale of two sources)
+
+The website that is built is a fully static site that can be viewed locally. It
```
It would be helpful to link to the other docs page for the local dashboard workflow here

zkamvar replied:
I always get confused trying to reason about [myst-parser's implicit cross-links](https://myst-parser.readthedocs.io/en/latest/syntax/cross-referencing.html#inline-links-with-implicit-text), so this may or may not work, but I'll fix it when I render this locally. 

```suggestion
The website that is built is a fully static site that [can be viewed locally](./dashboard-local.md). It
```

lshandross reviewed (changes requested):
A few comments from me, but generally looks good overall!

zkamvar commented:
Update: I've mad a small rearrangement in [f4bdc9c](https://github.com/hubverse-org/hubDocs/pull/326/commits/f4bdc9c9aee755c858214dd1be82eec91ad7a30d) that consolidates the explanation of the build process.

zkamvar commented:
I'm going to merge this as Li has provided approval and to favor moving forward.

@annakrystalli, I believe I have addressed your concerns, but if you see something, say something and I will address it.

zkamvar commented:
Thank you both for your reviews!
